Sources/ArcGISToolkit/Components/Augmented Reality/CalibrationView.swift

```diff
@@ -0,0 +1,170 @@
+        private var joystickHeadingDelta: Double {
+            Double(signOf: headingSliderValue, magnitudeOf: headingSliderValue * headingSliderValue / 25)
+        }
+        /// The elevation delta amount based on the elevation slider value.
+        private var joystickElevationDelta: Double {
+            Double(signOf: elevationSliderValue, magnitudeOf: elevationSliderValue * elevationSliderValue / 100)
+        }
+        
+        var body: some View {
+            Button {
+                viewModel.isCalibrating = true
+                setBasemapOpacity(0.5)
+            } label: {
+                Text("Calibrate")
+            }
+            .popover(isPresented: $viewModel.isCalibrating) {
+                VStack {
+                    Text("Heading: \(viewModel.calibrationHeading?.rounded(.towardZero) ?? viewModel.cameraController.originCamera.heading.rounded(.towardZero), format: .number)")
+                    
+                    HStack {
+                        Button {
+                            let heading = viewModel.cameraController.originCamera.heading - 1
+                            updateHeading(heading)
+                        } label: {
+                            Image(systemName: "minus")
+                        }
+                        Slider(value: $headingSliderValue, in: -10...10) { editingChanged in
+                            if !editingChanged {
+                                headingTimer?.invalidate()
+                                headingTimer = nil
+                                headingSliderValue = 0.0
+                            }
+                        }
+                        .onChange(of: headingSliderValue) { heading in
+                            guard headingTimer == nil else { return }
+                            // Create a timer which rotates the camera when fired.
+                            let timer = Timer(timeInterval: 0.1, repeats: true) { [self] (_) in
+                                rotateHeading(joystickHeadingDelta)
+                            }
+                            headingTimer = timer
+                            // Add the timer to the main run loop.
+                            RunLoop.main.add(timer, forMode: .default)
+                        }
+                        
+                        Button {
+                            let heading = viewModel.cameraController.originCamera.heading + 1
+                            updateHeading(heading)
+                        } label: {
+                            Image(systemName: "plus")
+                        }
+                    }
+                    VStack {
+                        Text("Elevation: \(viewModel.calibrationElevation?.rounded(.towardZero) ?? viewModel.cameraController.originCamera.location.z?.rounded(.towardZero) ?? 0, format: .number)")
+                        HStack {
+                            Button {
+                                updateElevation(-1)
```

Comment:
It may be worth retaining the argument label here. When I read this method it gives me an impression that it is updating the elevation value to `-1`

Reply:
I added argument labels `by` and `to` for clarity.

Reply:
I realized that the 2 `updateHeading(to:)` calls are replaceable by `rotateHeading(by:)`? For example, the add button `updateHeading(to: heading)` can be replaced by `rotateHeading(by: 1)`

Reply:
Good catch.